File: Arrays/Medium/C++/Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
                int n = prices.size();
        int maxRight = 0;
        int maxProfit = 0;
        for(int i = n - 1; i >= 0; i--){
            maxRight = max(prices[i], maxRight);
            maxProfit = max(maxProfit, maxRight - prices[i]);
        }
        
        return maxProfit;
        
    }
};

File: Arrays/Medium/Java/Best Time to Buy and Sell Stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;
        int maxRight = 0;
        int n = prices.length; 
        for(int i=n-1; i>=0; i--) {
            maxRight = Math.max(maxRight, prices[i]);
            maxProfit = Math.max(maxProfit, maxRight-prices[i]);
        }
        return maxProfit;
    }
}
